packages/device-management-kit/src/internal/manager-api/data/ManagerApiDataSource.ts

```diff
@@ -66,4 +73,51 @@ export interface ManagerApiDataSource {
    * Returns the current provider.
    */
   getProvider(): number;
+
+  /**
+   * Retrieves a firmware version from a final firmware ID.
+   *
+   * @param finalFirmwareId - The ID of the final firmware to retrieve.
+   * @returns EitherAsync containing the final firmware or an HttpFetchApiError.
+   */
+  getFirmwareVersionById(
+    finalFirmwareId: number,
+  ): EitherAsync<HttpFetchApiError, FinalFirmware>;
+
+  /**
+   * Retrieves the an OSU firmware version for a given version, device ID.
+   *
+   * @param params - The parameters for getting the firmware version.
+   * @returns EitherAsync containing the OSU firmware or an HttpFetchApiError.
+   */
+  getOsuFirmwareVersion(
+    params: GetFirmwareVersionParams,
+  ): EitherAsync<HttpFetchApiError, OsuFirmware>;
+
+  /**
+   * Retrieves the latest firmware available for a given current firmware, device ID.
+   *
+   * @param params - The parameters for getting the firmware version.
+   * @returns EitherAsync containing the OSU firmware or an HttpFetchApiError.
+   */
+  getLatestFirmwareVersion(
+    params: GetLatestFirmwareVersionParams,
+  ): EitherAsync<HttpFetchApiError, OsuFirmware>;
+
+  /**
+   * Retrieves the available language packages for a device.
+   *
+   * @param params - The parameters for getting the language packages.
+   * @returns EitherAsync containing the list of language packages or an HttpFetchApiError.
+   */
+  getLanguagePackages(
+    params: GetLanguagePackagesParams,
+  ): EitherAsync<HttpFetchApiError, LanguagePackage[]>;
```

Comment:
_[COULD]_ here and below
```suggestion
  ): EitherAsync<HttpFetchApiError, Array<LanguagePackage>>;
```